### PR TITLE
fix: Clean up JNI code and properly free strings

### PR DESCRIPTION
**Pull request by Swatinem**
## :loudspeaker: Type of change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring

## :scroll: Description

This cleans up the JNI layer, removes some unused things like the global transport options, and makes sure to clean up utf strings that have been created.

## :bulb: Motivation and Context

Avoids potential memory leaks and cleans up obsolete code.

## :pencil: Checklist

- [ ] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] All tests passing
- [ ] No breaking changes

### CODE REVIEW

**Comment by marandaneto** [comment]
looks good, missing a changelog and filling up the PR template `Motivation and Context`, is this related to the memory corruption thingie?

**Comment by Swatinem** [comment]
> is this related to the memory corruption thingie?

I noticed it when investigating, yes

**Comment by marandaneto** [comment]
> > is this related to the memory corruption thingie?
> 
> I noticed it when investigating, yes

niiice, thank you!

**Review by bruno-garcia — APPROVED** [review]
👀

**Comment by bruno-garcia** [comment]
Missing a changelog entry

I.e:
* Removing a ton of leaks

:trollface:

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@
 * Enhancement: Make SentryExceptionResolver Order configurable to not send handled web exceptions (#1008)
 * Enhancement: Resolve HTTP Proxy parameters from the external configuration (#1028)
 * Enhancement: Sentry NDK integration is compiled against default NDK version based on AGP's version
+* Fix: Avoids potential memory leaks on the NDK integration and cleans up obsolete code.
```
Comment (marandaneto) [inline CHANGELOG.md:10]: Fix: Clean up JNI code and properly free strings #1050

**Review by marandaneto — APPROVED** [review]
LGTM

**Pull request merged**